pages/posts/flight-of-a-byte.tsx

```diff
@@ -0,0 +1,120 @@
+                </p> 
+                <p>
+                Without a central server, each group member may add to the message log and receive updates from others in a different order. But the synchronization algorithm ensures “eventual consistency”: after exchanging enough messages, eventually all the users will have the same message log in the same order. At that point they will all have the same “head” message, which refers to all the previous messages in a chain. Wesh currently uses OrbitDB for this, and you can read the details in <a href="https://github.com/orbitdb/field-manual/blob/main/03_The_Architecture_of_OrbitDB/02_ipfs-log.md" target="_blank" rel="noreferrer">their manual</a>. (In the future, Wesh may use a simpler library but it will still have the same concept of a synchronized message log with a head.)
+                </p>
+                <p>
+                 As mentioned, <code>client2</code> doesn&apos;t send the message “Hello” directly to the other group members. Instead, it puts the message in its own message log and updates the head. The other members learn about the new head which means there is a new message to fetch.
+                </p>
+                <h3>libp2p pubsub</h3>
+                <p>
+                 So now the question is, how does <code>client1</code> learn that <code>client2</code> says there is a new message log head? This uses libp2p pubsub where peers can subscribe to a topic and publish a pubsub message to that topic, or receive pubsub messages. (We say “pubsub message” to not confuse with the ordinary messages in the log which we&apos;ll discuss below.)
+                </p>
+                <p>
+                libp2p pubsub is a powerful tool with algorithms which allow it to scale to many peers and to make sure they all get the pubsub message. You can read more in <a href="https://docs.libp2p.io/concepts/pubsub/overview/" target="_blank" rel="noreferrer">its documentation</a>, but for this blog post we only need to say that the Wesh group has a libp2p topic name like 
+                &nbsp;<code>
+                /orbitdb/bafyreihpdptqwgpuwu4ob6nusxebmdx6fm6mq5pma5hva6eu5u74pflo7e/2ac4e88c5272f900c76fd36989f9780e5b2c95d75d38fe5bcef0345b34bc4806_wesh_group_messages
+                    </code>
+                &nbsp;a pubsub message to this topic that contains the message log head. The libp2p on <code>client1</code> receives this pubsub message and sends it (<a href="https://github.com/libp2p/go-libp2p-pubsub/blob/6d73cd4b56bf9d7fdd0c0caaed629a3d51027a75/pubsub.go#L978" target="_blank" rel="noreferrer">code</a>) internally to all processes which are subscribed to the topic. The Wesh application of <code>client1</code> has used ActivateGroup and is therefore subscribed to this topic and receives the pubsub message (<a href="https://github.com/berty/go-orbit-db/blob/368618e06b3be3bd17952d167d0e2f4fa03960ff/pubsub/pubsubraw/pubsub.go#L76" target="_blank" rel="noreferrer">code</a>).
```

Comment:
GroupMessageList is in a code block, but ActivateGroup not.

Reply:
I agree. For consistency, can you put ActivateGroup in code font.